azurerm/internal/services/apimanagement/api_management_product_api_resource.go

```diff
@@ -43,13 +45,15 @@ func resourceArmApiManagementProductApi() *schema.Resource {
 
 func resourceArmApiManagementProductApiCreate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).ApiManagement.ProductApisClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	resourceGroup := d.Get("resource_group_name").(string)
 	serviceName := d.Get("api_management_name").(string)
 	apiName := d.Get("api_name").(string)
 	productId := d.Get("product_id").(string)
+	id := parse.NewProductApiID(subscriptionId, resourceGroup, serviceName, productId, apiName)
```

Comment:
I don't think we can do this in the create function at this time, as if there is a change to the case (correcting it) it would change the ID and require a state migration function?

Reply:
What do you want me to do, remove this from create or add a state migration?

Reply:
OTOH why would it matter on create? I set the resource ID below from the generator, so kinda ignoring what API returns back..

Reply:
If another resource is depending on the ID being a certain/existing casing this could break resources that consume it. For now we are updating read/delete/update (but not resource ID) and tests to use the new parsers. Once every resource everywhere has been updated we can then move onto having the create functions use them.

Reply:
Resolved.